### ISSUE TITLE
QoE metrics report: Populate `QoEReport@recordingSessionId` attribute with media streaming session identifier

### ISSUE THREAD

**Issue by rjb1000** [5G-MAG/rt-5gms-media-session-handler#33]
_This is a request for an optional but useful feature of QoE metrics reporting to be implemented by the Media Session Handler._

# Feature description

The `urn:3gpp:metadata:2011:HSD:receptionreport` schema specified in clause 10.6.2 of TS 26.247 defines an optional attribute `QoEReport@recordingSessionId`.

Clause 11.4.2 of TS 26.512 does not (yet) require this optional attribute to be populated when the metrics report format `urn:3GPP:ns:PSS:DASH:QM10` is provisioned. However, it would be nice if the Media Session Handler could supply a value representing a **media streaming session identifier** in line with the proposal in 5G-MAG/Standards#82, **so that QoE metrics reports belonging to the same media streaming session can be collated together for analysis**.

I propose that the Media Session Handler generates a UUID for each new media streaming session and populates the `QoEReport@recordingSessionId` with this value (minus hyphens to satisfy the `hexBinary` XML schema type). As well as complying with the schema syntax in TS 26.247 clause 10.6.2, this is also within the spirit of the attribute quoted below. However, it does violate a length restriction on the attribute value imposed for use in the historical Packet-switched Streaming System (PSS), which could be changed for the 5GMS System in a future revision of TS 26.512, based on the proposal in 5G-MAG/Standards#82.

> The `recordingSessionId` is a **two-byte octet** defined by the client. **It shall remain the same for all QoE reports belonging to the same streaming session, and it should be different for QoE reports belonging to different streaming sessions.**

The second sentence explains that the session identifier should be different from each streaming session. Reading across to 5G Media Streaming, each new media streaming session should have a distinct UUID.

# Additional context

The reason it would be good to add this feature is that it forces us to think about the concept of a media streaming session as a first class logical entity in the Media Session Handler. This raises questions such as:

1. How does the Media Session Handler detect the start/end of a media streaming session?
1. What happens if there is more than one media streaming session active at the same time?
    - e.g., two media playback sessions running simultaneously in the same application view.
    - e.g., two different media playback applications running simultaneously on the same UE in multi-window mode or [multi-screen/desktop mode](https://www.samsung.com/uk/apps/samsung-dex/).
1. If there are multiple media streaming sessions active using the same Provisioning Session identifier, can they share the same Service Access Information to avoid having to contact the 5GMS AF separately at M5 for each session?
   - Can the retrieval of (and periodic validity check for) Service Access Information be factored out into a shared Media Session Handler cache so that two media streaming sessions referencing the same Provisioning Session running in parallel (or serially in quick succession) can take advantage of a cached resource?
1. But, conversely, how does the Media Session Handler provide different media streaming sessions with their own dedicated Network Assistance sessions and/or dynamic policies, as well an independent QoE metrics reporting and consumption reporting contexts?
1. How can the Media Session Handler provide adequate isolation between the contexts of different media streaming sessions to prevent access to sensitive user data (e.g. consumption reporting) belonging to one session by all other active sessions?

**Comment by dsilhavy** [comment]
Added support for `recordingSessionId`. In our implementation, all requests to start a new streaming session go through the Media Session Handler. The Media Session Handler issues a playback request to the Media Stream Handler providing a list of possible endpoints and additional metadata. This metadata was enhanced to include a `mediaStreamingSessionIdentifier`:

````kotlin
data class PlaybackRequest(
    val entryPoints: ArrayList<EntryPoint>,
    val consumptionRequest: ConsumptionRequest,
    val qoeMetricsRequests: ArrayList<QoeMetricsRequest>,
    val mediaStreamingSessionIdentifier: String
) : Parcelable
````

The Media Stream Handler uses that `mediaStreamingSessionIdentifier` to populate the `QoEReport@recordingSessionId`. As an example:

## Streaming Session 1

### QoE Metrics Report - 1

````xml
<?xml version="1.0" encoding="utf-8"?>
<ReceptionReport clientID="0e9e707b-0dfe-4449-99ce-01ee4e2ad4db"
                 contentURI="https://livesim.dashif.org/livesim/testpic_2s/Manifest.mpd"
                 xsi:schemaLocation="urn:3gpp:metadata:2011:HSD:receptionreport DASH-QoE-Report.xsd"
                 xmlns:sv="urn:3gpp:metadata:2016:PSS:schemaVersion" xmlns="urn:3gpp:metadata:2011:HSD:receptionreport"
                 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    <QoeReport recordingSessionId="ab8cab64d3704b5a9981f6673e73516a" reportPeriod="10"
               reportTime="2024-03-15T10:21:11.064Z" periodID="P0">
        <QoeMetric>
            <BufferLevel>
                <BufferLevelEntry level="0" t="2024-03-15T10:21:07.638Z"/>
                <BufferLevelEntry level="0" t="2024-03-15T10:21:07.764Z"/>
                <BufferLevelEntry level="0" t="2024-03-15T10:21:08.416Z"/>
                <BufferLevelEntry level="0" t="2024-03-15T10:21:08.716Z"/>
                <BufferLevelEntry level="0" t="2024-03-15T10:21:09.037Z"/>
                <BufferLevelEntry level="0" t="2024-03-15T10:21:09.152Z"/>
                <BufferLevelEntry level="0" t="2024-03-15T10:21:09.198Z"/>
                <BufferLevelEntry level="2000" t="2024-03-15T10:21:09.351Z"/>
                <BufferLevelEntry level="2000" t="2024-03-15T10:21:09.362Z"/>
                <BufferLevelEntry level="3984" t="2024-03-15T10:21:09.498Z"/>
                <BufferLevelEntry level="3983" t="2024-03-15T10:21:09.505Z"/>
            </BufferLevel>
        </QoeMetric>
        <QoeMetric>
            <HttpList>
                <HttpListEntry actualurl="https://livesim.dashif.org/livesim2/testpic_2s/Manifest.mpd" interval="763"
                               range="" responsecode="200" trequest="2024-03-15T10:21:07.645Z"
                               tresponse="2024-03-15T10:21:07.649Z" type="MPD"
                               url="https://livesim.dashif.org/livesim2/testpic_2s/Manifest.mpd">
                    <Trace b="2121" d="763" s="2024-03-15T10:21:07.649Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://time.akamai.com/?iso&amp;ms" interval="292" range=""
                               responsecode="200" trequest="2024-03-15T10:21:08.415Z"
                               tresponse="2024-03-15T10:21:08.421Z" type="other"
                               url="https://time.akamai.com/?iso&amp;ms">
                    <Trace b="24" d="292" s="2024-03-15T10:21:08.421Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://livesim.dashif.org/livesim2/testpic_2s/V300/init.mp4" interval="68"
                               range="" responsecode="200" trequest="2024-03-15T10:21:08.961Z"
                               tresponse="2024-03-15T10:21:08.966Z" type="InitializationSegment"
                               url="https://livesim.dashif.org/livesim2/testpic_2s/V300/init.mp4">
                    <Trace b="678" d="68" s="2024-03-15T10:21:08.966Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://livesim.dashif.org/livesim2/testpic_2s/V300/855249019.m4s"
                               interval="108" range="" responsecode="200" trequest="2024-03-15T10:21:09.039Z"
                               tresponse="2024-03-15T10:21:09.042Z" type="MediaSegment"
                               url="https://livesim.dashif.org/livesim2/testpic_2s/V300/855249019.m4s">
                    <Trace b="40560" d="108" s="2024-03-15T10:21:09.042Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://livesim.dashif.org/livesim2/testpic_2s/A48/init.mp4" interval="226"
                               range="" responsecode="200" trequest="2024-03-15T10:21:08.964Z"
                               tresponse="2024-03-15T10:21:08.967Z" type="InitializationSegment"
                               url="https://livesim.dashif.org/livesim2/testpic_2s/A48/init.mp4">
                    <Trace b="614" d="226" s="2024-03-15T10:21:08.967Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://livesim.dashif.org/livesim2/testpic_2s/A48/855249019.m4s"
                               interval="142" range="" responsecode="200" trequest="2024-03-15T10:21:09.193Z"
                               tresponse="2024-03-15T10:21:09.198Z" type="MediaSegment"
                               url="https://livesim.dashif.org/livesim2/testpic_2s/A48/855249019.m4s">
                    <Trace b="13120" d="142" s="2024-03-15T10:21:09.198Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://livesim.dashif.org/livesim2/testpic_2s/V300/855249020.m4s"
                               interval="138" range="" responsecode="200" trequest="2024-03-15T10:21:09.220Z"
                               tresponse="2024-03-15T10:21:09.222Z" type="MediaSegment"
                               url="https://livesim.dashif.org/livesim2/testpic_2s/V300/855249020.m4s">
                    <Trace b="39743" d="138" s="2024-03-15T10:21:09.222Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://livesim.dashif.org/livesim2/testpic_2s/A48/855249020.m4s"
                               interval="129" range="" responsecode="200" trequest="2024-03-15T10:21:09.364Z"
                               tresponse="2024-03-15T10:21:09.366Z" type="MediaSegment"
                               url="https://livesim.dashif.org/livesim2/testpic_2s/A48/855249020.m4s">
                    <Trace b="13253" d="129" s="2024-03-15T10:21:09.366Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://livesim.dashif.org/livesim2/testpic_2s/V300/855249021.m4s"
                               interval="125" range="" responsecode="200" trequest="2024-03-15T10:21:09.377Z"
                               tresponse="2024-03-15T10:21:09.379Z" type="MediaSegment"
                               url="https://livesim.dashif.org/livesim2/testpic_2s/V300/855249021.m4s">
                    <Trace b="39508" d="125" s="2024-03-15T10:21:09.379Z"/>
                </HttpListEntry>
            </HttpList>
        </QoeMetric>
        <QoeMetric>
            <MPDInformation representationId="V300">
                <Mpdinfo bandwidth="300000" codecs="avc1.64001e" frameRate="30.0" height="360" mimeType="video/mp4"
                         width="640"/>
            </MPDInformation>
            <MPDInformation representationId="A48">
                <Mpdinfo bandwidth="48000" codecs="mp4a.40.2" mimeType="audio/mp4"/>
            </MPDInformation>
        </QoeMetric>
        <QoeMetric>
            <RepSwitchList>
                <RepSwitchEvent mt="P0Y0DT0H0M30S" t="2024-03-15T10:21:09.062Z" to="V300"/>
                <RepSwitchEvent mt="P0Y0DT0H0M30S" t="2024-03-15T10:21:09.222Z" to="A48"/>
            </RepSwitchList>
        </QoeMetric>
        <sv:delimiter>0</sv:delimiter>
    </QoeReport>
</ReceptionReport>
````

### QoE Metrics Report - 2

````xml
<?xml version="1.0" encoding="utf-8"?>
<ReceptionReport clientID="0e9e707b-0dfe-4449-99ce-01ee4e2ad4db"
                 contentURI="https://livesim.dashif.org/livesim/testpic_2s/Manifest.mpd"
                 xsi:schemaLocation="urn:3gpp:metadata:2011:HSD:receptionreport DASH-QoE-Report.xsd"
                 xmlns:sv="urn:3gpp:metadata:2016:PSS:schemaVersion" xmlns="urn:3gpp:metadata:2011:HSD:receptionreport"
                 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    <QoeReport recordingSessionId="ab8cab64d3704b5a9981f6673e73516a" reportPeriod="10"
               reportTime="2024-03-15T10:21:31.403Z" periodID="P0">
        <QoeMetric>
            <BufferLevel>
                <BufferLevelEntry level="29774" t="2024-03-15T10:21:21.770Z"/>
                <BufferLevelEntry level="30408" t="2024-03-15T10:21:22.129Z"/>
                <BufferLevelEntry level="31398" t="2024-03-15T10:21:22.145Z"/>
                <BufferLevelEntry level="29795" t="2024-03-15T10:21:23.757Z"/>
                <BufferLevelEntry level="29462" t="2024-03-15T10:21:24.108Z"/>
                <BufferLevelEntry level="31375" t="2024-03-15T10:21:24.163Z"/>
                <BufferLevelEntry level="29777" t="2024-03-15T10:21:25.768Z"/>
                <BufferLevelEntry level="29430" t="2024-03-15T10:21:26.147Z"/>
                <BufferLevelEntry level="31375" t="2024-03-15T10:21:26.168Z"/>
                <BufferLevelEntry level="29927" t="2024-03-15T10:21:27.614Z"/>
                <BufferLevelEntry level="29765" t="2024-03-15T10:21:27.781Z"/>
                <BufferLevelEntry level="29475" t="2024-03-15T10:21:28.103Z"/>
                <BufferLevelEntry level="30455" t="2024-03-15T10:21:28.124Z"/>
                <BufferLevelEntry level="29791" t="2024-03-15T10:21:29.760Z"/>
                <BufferLevelEntry level="29442" t="2024-03-15T10:21:30.132Z"/>
                <BufferLevelEntry level="31379" t="2024-03-15T10:21:30.165Z"/>
            </BufferLevel>
        </QoeMetric>
        <QoeMetric>
            <HttpList>
                <HttpListEntry actualurl="https://livesim.dashif.org/livesim2/testpic_2s/Manifest.mpd" interval="114"
                               range="" responsecode="200" trequest="2024-03-15T10:21:21.654Z"
                               tresponse="2024-03-15T10:21:21.655Z" type="MPD"
                               url="https://livesim.dashif.org/livesim2/testpic_2s/Manifest.mpd">
                    <Trace b="2121" d="114" s="2024-03-15T10:21:21.655Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://livesim.dashif.org/livesim2/testpic_2s/A48/855249040.m4s"
                               interval="86" range="" responsecode="200" trequest="2024-03-15T10:21:22.037Z"
                               tresponse="2024-03-15T10:21:22.041Z" type="MediaSegment"
                               url="https://livesim.dashif.org/livesim2/testpic_2s/A48/855249040.m4s">
                    <Trace b="13314" d="86" s="2024-03-15T10:21:22.041Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://livesim.dashif.org/livesim2/testpic_2s/V300/855249040.m4s"
                               interval="114" range="" responsecode="200" trequest="2024-03-15T10:21:22.027Z"
                               tresponse="2024-03-15T10:21:22.029Z" type="MediaSegment"
                               url="https://livesim.dashif.org/livesim2/testpic_2s/V300/855249040.m4s">
                    <Trace b="40559" d="114" s="2024-03-15T10:21:22.029Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://livesim.dashif.org/livesim2/testpic_2s/Manifest.mpd" interval="94"
                               range="" responsecode="200" trequest="2024-03-15T10:21:23.655Z"
                               tresponse="2024-03-15T10:21:23.659Z" type="MPD"
                               url="https://livesim.dashif.org/livesim2/testpic_2s/Manifest.mpd">
                    <Trace b="2121" d="94" s="2024-03-15T10:21:23.659Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://livesim.dashif.org/livesim2/testpic_2s/A48/855249041.m4s"
                               interval="82" range="" responsecode="200" trequest="2024-03-15T10:21:24.022Z"
                               tresponse="2024-03-15T10:21:24.024Z" type="MediaSegment"
                               url="https://livesim.dashif.org/livesim2/testpic_2s/A48/855249041.m4s">
                    <Trace b="13292" d="82" s="2024-03-15T10:21:24.024Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://livesim.dashif.org/livesim2/testpic_2s/V300/855249041.m4s"
                               interval="137" range="" responsecode="200" trequest="2024-03-15T10:21:24.019Z"
                               tresponse="2024-03-15T10:21:24.023Z" type="MediaSegment"
                               url="https://livesim.dashif.org/livesim2/testpic_2s/V300/855249041.m4s">
                    <Trace b="39465" d="137" s="2024-03-15T10:21:24.023Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://livesim.dashif.org/livesim2/testpic_2s/Manifest.mpd" interval="107"
                               range="" responsecode="200" trequest="2024-03-15T10:21:25.655Z"
                               tresponse="2024-03-15T10:21:25.658Z" type="MPD"
                               url="https://livesim.dashif.org/livesim2/testpic_2s/Manifest.mpd">
                    <Trace b="2121" d="107" s="2024-03-15T10:21:25.658Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://livesim.dashif.org/livesim2/testpic_2s/A48/855249042.m4s"
                               interval="116" range="" responsecode="200" trequest="2024-03-15T10:21:26.028Z"
                               tresponse="2024-03-15T10:21:26.030Z" type="MediaSegment"
                               url="https://livesim.dashif.org/livesim2/testpic_2s/A48/855249042.m4s">
                    <Trace b="13218" d="116" s="2024-03-15T10:21:26.030Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://livesim.dashif.org/livesim2/testpic_2s/V300/855249042.m4s"
                               interval="144" range="" responsecode="200" trequest="2024-03-15T10:21:26.020Z"
                               tresponse="2024-03-15T10:21:26.022Z" type="MediaSegment"
                               url="https://livesim.dashif.org/livesim2/testpic_2s/V300/855249042.m4s">
                    <Trace b="39241" d="144" s="2024-03-15T10:21:26.022Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://livesim.dashif.org/livesim2/testpic_2s/Manifest.mpd" interval="122"
                               range="" responsecode="200" trequest="2024-03-15T10:21:27.656Z"
                               tresponse="2024-03-15T10:21:27.658Z" type="MPD"
                               url="https://livesim.dashif.org/livesim2/testpic_2s/Manifest.mpd">
                    <Trace b="2121" d="122" s="2024-03-15T10:21:27.658Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://livesim.dashif.org/livesim2/testpic_2s/A48/855249043.m4s"
                               interval="72" range="" responsecode="200" trequest="2024-03-15T10:21:28.026Z"
                               tresponse="2024-03-15T10:21:28.028Z" type="MediaSegment"
                               url="https://livesim.dashif.org/livesim2/testpic_2s/A48/855249043.m4s">
                    <Trace b="13170" d="72" s="2024-03-15T10:21:28.028Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://livesim.dashif.org/livesim2/testpic_2s/V300/855249043.m4s"
                               interval="104" range="" responsecode="200" trequest="2024-03-15T10:21:28.018Z"
                               tresponse="2024-03-15T10:21:28.019Z" type="MediaSegment"
                               url="https://livesim.dashif.org/livesim2/testpic_2s/V300/855249043.m4s">
                    <Trace b="39466" d="104" s="2024-03-15T10:21:28.019Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://livesim.dashif.org/livesim2/testpic_2s/Manifest.mpd" interval="90"
                               range="" responsecode="200" trequest="2024-03-15T10:21:29.659Z"
                               tresponse="2024-03-15T10:21:29.664Z" type="MPD"
                               url="https://livesim.dashif.org/livesim2/testpic_2s/Manifest.mpd">
                    <Trace b="2121" d="90" s="2024-03-15T10:21:29.664Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://livesim.dashif.org/livesim2/testpic_2s/A48/855249044.m4s"
                               interval="88" range="" responsecode="200" trequest="2024-03-15T10:21:30.040Z"
                               tresponse="2024-03-15T10:21:30.042Z" type="MediaSegment"
                               url="https://livesim.dashif.org/livesim2/testpic_2s/A48/855249044.m4s">
                    <Trace b="13327" d="88" s="2024-03-15T10:21:30.042Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://livesim.dashif.org/livesim2/testpic_2s/V300/855249044.m4s"
                               interval="128" range="" responsecode="200" trequest="2024-03-15T10:21:30.033Z"
                               tresponse="2024-03-15T10:21:30.035Z" type="MediaSegment"
                               url="https://livesim.dashif.org/livesim2/testpic_2s/V300/855249044.m4s">
                    <Trace b="39735" d="128" s="2024-03-15T10:21:30.035Z"/>
                </HttpListEntry>
            </HttpList>
        </QoeMetric>
        <sv:delimiter>0</sv:delimiter>
    </QoeReport>
</ReceptionReport>
````

## Streaming Session 2 (change of source)

### QoE Metrics Report - 1

````xml
<?xml version="1.0" encoding="utf-8"?>
<ReceptionReport clientID="0e9e707b-0dfe-4449-99ce-01ee4e2ad4db"
                 contentURI="https://dash.akamaized.net/envivio/EnvivioDash3/manifest.mpd"
                 xsi:schemaLocation="urn:3gpp:metadata:2011:HSD:receptionreport DASH-QoE-Report.xsd"
                 xmlns:sv="urn:3gpp:metadata:2016:PSS:schemaVersion" xmlns="urn:3gpp:metadata:2011:HSD:receptionreport"
                 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    <QoeReport recordingSessionId="a40b185177aa45babac70508b6f6f667" reportPeriod="10"
               reportTime="2024-03-15T10:21:39.560Z" periodID="period0">
        <QoeMetric>
            <BufferLevel>
                <BufferLevelEntry level="0" t="2024-03-15T10:21:38.574Z"/>
                <BufferLevelEntry level="0" t="2024-03-15T10:21:38.584Z"/>
                <BufferLevelEntry level="0" t="2024-03-15T10:21:38.936Z"/>
                <BufferLevelEntry level="0" t="2024-03-15T10:21:39.054Z"/>
                <BufferLevelEntry level="0" t="2024-03-15T10:21:39.125Z"/>
                <BufferLevelEntry level="520" t="2024-03-15T10:21:39.184Z"/>
                <BufferLevelEntry level="840" t="2024-03-15T10:21:39.239Z"/>
                <BufferLevelEntry level="1960" t="2024-03-15T10:21:39.309Z"/>
                <BufferLevelEntry level="2240" t="2024-03-15T10:21:39.396Z"/>
            </BufferLevel>
        </QoeMetric>
        <QoeMetric>
            <HttpList>
                <HttpListEntry actualurl="https://dash.akamaized.net/envivio/EnvivioDash3/manifest.mpd" interval="357"
                               range="" responsecode="200" trequest="2024-03-15T10:21:38.577Z"
                               tresponse="2024-03-15T10:21:38.578Z" type="MPD"
                               url="https://dash.akamaized.net/envivio/EnvivioDash3/manifest.mpd">
                    <Trace b="2726" d="357" s="2024-03-15T10:21:38.578Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://dash.akamaized.net/envivio/EnvivioDash3/v7_257-Header.m4s"
                               interval="40" range="" responsecode="200" trequest="2024-03-15T10:21:39.013Z"
                               tresponse="2024-03-15T10:21:39.014Z" type="InitializationSegment"
                               url="https://dash.akamaized.net/envivio/EnvivioDash3/v7_257-Header.m4s">
                    <Trace b="688" d="40" s="2024-03-15T10:21:39.014Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://dash.akamaized.net/envivio/EnvivioDash3/v4_258-Header.m4s"
                               interval="108" range="" responsecode="200" trequest="2024-03-15T10:21:39.015Z"
                               tresponse="2024-03-15T10:21:39.016Z" type="InitializationSegment"
                               url="https://dash.akamaized.net/envivio/EnvivioDash3/v4_258-Header.m4s">
                    <Trace b="615" d="108" s="2024-03-15T10:21:39.016Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://dash.akamaized.net/envivio/EnvivioDash3/v4_258-270146-i-1.m4s"
                               interval="45" range="" responsecode="200" trequest="2024-03-15T10:21:39.138Z"
                               tresponse="2024-03-15T10:21:39.138Z" type="MediaSegment"
                               url="https://dash.akamaized.net/envivio/EnvivioDash3/v4_258-270146-i-1.m4s">
                    <Trace b="33578" d="45" s="2024-03-15T10:21:39.138Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://dash.akamaized.net/envivio/EnvivioDash3/v4_258-270146-i-2.m4s"
                               interval="55" range="" responsecode="200" trequest="2024-03-15T10:21:39.182Z"
                               tresponse="2024-03-15T10:21:39.183Z" type="MediaSegment"
                               url="https://dash.akamaized.net/envivio/EnvivioDash3/v4_258-270146-i-2.m4s">
                    <Trace b="32948" d="55" s="2024-03-15T10:21:39.183Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://dash.akamaized.net/envivio/EnvivioDash3/v7_257-270146-i-1.m4s"
                               interval="239" range="" responsecode="200" trequest="2024-03-15T10:21:39.069Z"
                               tresponse="2024-03-15T10:21:39.070Z" type="MediaSegment"
                               url="https://dash.akamaized.net/envivio/EnvivioDash3/v7_257-270146-i-1.m4s">
                    <Trace b="1535564" d="239" s="2024-03-15T10:21:39.070Z"/>
                </HttpListEntry>
                <HttpListEntry actualurl="https://dash.akamaized.net/envivio/EnvivioDash3/v4_258-270146-i-3.m4s"
                               interval="66" range="" responsecode="200" trequest="2024-03-15T10:21:39.329Z"
                               tresponse="2024-03-15T10:21:39.330Z" type="MediaSegment"
                               url="https://dash.akamaized.net/envivio/EnvivioDash3/v4_258-270146-i-3.m4s">
                    <Trace b="32770" d="66" s="2024-03-15T10:21:39.330Z"/>
                </HttpListEntry>
            </HttpList>
        </QoeMetric>
        <QoeMetric>
            <MPDInformation representationId="v7_257">
                <Mpdinfo bandwidth="5300000" codecs="avc1.4D401E" frameRate="29.970029830932617" height="1080"
                         mimeType="video/mp4" width="1920"/>
            </MPDInformation>
            <MPDInformation representationId="v4_258">
                <Mpdinfo bandwidth="130800" codecs="mp4a.40.2" mimeType="audio/mp4"/>
            </MPDInformation>
        </QoeMetric>
        <QoeMetric>
            <RepSwitchList>
                <RepSwitchEvent mt="P0Y0DT0H0M0S" t="2024-03-15T10:21:39.089Z" to="v7_257"/>
                <RepSwitchEvent mt="P0Y0DT0H0M0S" t="2024-03-15T10:21:39.145Z" to="v4_258"/>
            </RepSwitchList>
        </QoeMetric>
        <sv:delimiter>0</sv:delimiter>
    </QoeReport>
</ReceptionReport>
````

**Comment by dsilhavy** [comment]
Addressed in #56
